### ISSUE TITLE
Website shows coreError500Title on Windows

### ISSUE THREAD

**Issue by JetGum** [datenstrom/yellow#533]
Hi all,

when updating Core 0.8.13 to 0.8.14, my site breaks.

After Core update:
![image](https://user-images.githubusercontent.com/48367334/89125616-06332780-d4e0-11ea-9aa1-7b7bedcb14ac.png)

After just updating berlin.php, my customized theme gets overwritten.

On an installation of yellow-master, things get ugly when updating from 0.8.13. to 0.8.14:
![image](https://user-images.githubusercontent.com/48367334/89125695-9c674d80-d4e0-11ea-82ea-1167898f15ec.png)

Even a fresh installation of Yellow master 0.8.14 gives me:
![image](https://user-images.githubusercontent.com/48367334/89126740-6a59e980-d4e8-11ea-9a90-1103000ebb75.png)

To me this is rather cryptic, so i don't know where to start to remedy this. 

Cheers,

Carsten

**Comment by markseu** [comment]
Updating the core extension manually is not a good idea. Datenstrom Yellow uses a rolling release model, this means you have to update all extensions at once. In order to fix your website, I would restore the `core.php` from your trash folder and then run the normal website update. Hope this helps.

**Comment by JetGum** [comment]
Hi Mark, 

this seems to be a localhost issue (s. Screenshots above),, but I still would be glad if I could track the issue down and fix it.

First thing I did on **localhost** was the "update all extensions at once" thing, and it broke my site. Only after that I tried individually with Core.php, berlin.php etc. which yielded the same result. 

On live server, I ran an update on one site to 0.8.14 and it went smoothly. When I downloaded this site from my live server to localhost, coreError is back. so there seems to be an issue with Xampp (v3.2.3), local firewall, Apache confoguration or whatever.

Cheers,

**Comment by markseu** [comment]
I can't reproduce the problem on Mac and Linux. Can you do a fresh installation and show your `system/extensions/yellow.log`? Have any additional information what could cause this problem?

**Comment by JetGum** [comment]
I did a fresh install of yellow master, Downloaded, unzipped and went straight to localhost/yellow-master... [coreErrorTitle]

This is the log:
2020-08-06 20:02:43 info Datenstrom Yellow 2020.6, PHP 7.4.8, Apache 2.4.43, WINNT
2020-08-06 20:02:43 info Install extension 'English 0.8.23'
2020-08-06 20:02:43 info Install extension 'French 0.8.23'
2020-08-06 20:02:43 info Install extension 'German 0.8.23'

The Apache access.log says simply (apart from some 200 status code:

::1 - - [06/Aug/2020:22:04:25 +0200] "GET /yellow-master/ HTTP/1.1" **500** 1158 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/84.0.4147.105 Safari/537.36"

**Comment by JetGum** [comment]
codeError still persists with a fresh install of Xampp v3.2.4 / PHP 7.3.20 and Yellow 0.8.15

Yellow.log says:
2020-08-07 05:20:00 info Datenstrom Yellow 0.8.15, PHP 7.3.20, Apache 2.4.43, WINNT
2020-08-07 05:20:00 info Install extension 'English 0.8.23'
2020-08-07 05:20:00 info Install extension 'French 0.8.23'
2020-08-07 05:20:00 info Install extension 'German 0.8.23'

**Comment by schulle4u** [comment]
I can confirm the issue. However I think this is not a Yellow-related problem. It looks like Xampp misses some php modules that Yellow requires to run correctly. Unfortunately I wasn't able to track down the problem as Apache2 fails to write a propper error.log. Maybe insufficient write permissions in the installation folder. 

As a side note you should consider switching to a better development environment. Even the Windows subsystem for linux should work much better than Xampp. Furthermore Microsoft recently announced that they will drop Windows support for php with v8.0, support for older php versions will end depending on their life cycle.

**Comment by JetGum** [comment]
Ok, the just 2 more questions: 
Will live Apache servers be affected by this? And what would be a better development environment?

**Comment by JetGum** [comment]
Any ideads as to which php modules are missing? It's strange that some part of the markdown is rendered (header, navigation, footer), but contents not:

![image](https://user-images.githubusercontent.com/48367334/89688661-f3b35680-d902-11ea-84ed-0cb53a4d21f7.png)

**Comment by schulle4u** [comment]
According to our tests your live site won't be affected and it should be safe to upgrade, this looks like a windows/xampp related problem. I'm afraid to say that Windows support for Yellow currently has rather low priority for us. 

I'm using a Raspberry Pi model 3B+ with Raspbian and standard Apache/PHP packages as my testing environment. You can also try WSL (Windows subsystem for linux), I'm using this as my development environment for publishing extensions to GitHub. However the WSL sometimes lags performance, so it might not the best solution to run a full-blown LAMP stack.

**Comment by JetGum** [comment]
I have followed your advice and activated WSL, installed Ubuntu 18.4 LTS and Apache/PHP. Everything runs smoothly so far. PHP is being parsed: I see Apache2 Ubuntu Default Page and phpinfo on locahost in my browser. Visual Studio works as well, no probs here.

Only problem: I only see the files in my Yellow Directory when I open up Yellow on localhost (Yellow Master sits in C:/Sites/yellow, which translates into /mnt/c/Sites/yellow on WSL.)

I searched for tutorials (found [this ](https://websiteforstudents.com/install-datenstrom-yellow-cms-on-ubuntu-16-04-18-04-18-10-with-apache2-php-7-2/)and [this](https://medium.com/@fiqriismail/how-to-setup-apache-mysql-and-php-in-linux-subsystem-for-windows-10-e03e67afe6ee)), but still Yellow does not come up on localhost/yellow, only the files...

Does anyone have a link to another tutorial?

Cheers, and thanks for your support and patience so far.

**Comment by schulle4u** [comment]
You probably have to modify the default directory settings in your `/etc/apache2/apache2.conf`. 

```
<Directory /var/www/>
    Options Indexes FollowSymLinks
    AllowOverride All
    Require all granted
</Directory>
```

This should make the Yellow `.htaccess` directives work.

**Comment by JetGum** [comment]
Thx, looks like we're getting closer...

I checked mod_rewrite and it was already enabled, I only had to change "AllowOverride None" to "All" in the .conf file you mentioned.

Now Yellow is parsed on localhost... but I see 
"Server error. Something went wrong. Can't write file 'system/extensions/bundle-684b0ca552.min.css'!"

Looks like write access is not there, 

I tried [this](https://datenstrom.se/yellow/help/troubleshooting) from the helpfiles (chmod -R a+rw *) both in /Sites/ and in /Sites/yellow, but nope..

**Comment by schulle4u** [comment]
The project folder should belong to the appropriate webserver user, www-data by default. Try a `chown -r www-data:www-data yellow/`

**Comment by JetGum** [comment]
Only `sudo chown -R www-data:www-data yellow/` runs, but does not change the user to www-data. Also: 'sudo chown -R www-data:www-data /mnt/c/Sites/yellow'

drwxrwxrwx 1 jetgum jetgum 512 Aug 8 17:40 yellow

**Comment by schulle4u** [comment]
OK, looks like WSL doesn't fully support external directory permissions. I haven't triet this myself, but you can try it the other way around. 

1. Open Bash prompt and delete your symlink for the Yellow project folder. 
2. Leave your bash prompt opened, press Win+R and Type: `\\wsl$`
3. Navigate to your webroot directory in the explorer window, e. g. `Ubuntu\var\www\html`. 
4. Copy your project folder from `C:\` into your webroot. 
5. Adjust permissions and ownership from the bash prompt if applicable. 
6. Create/adjust your VS Code project's working directory, e. g. `\\wsl$\Ubuntu\var\www\html\yellow\` (never used VS code, don't know if it supports this kind of path).


HTH 😅

**Comment by JetGum** [comment]
Uh, interesting approach, haven't tried it yet because... somehow "www-data" is now owner of my project folder! Hoooraaayy!!
After 6-7 hours of trial & error and your relentless support... I see the Yellow frontend, and I can login into the backend, too. I can update and I have all my projects in one place.

Uff. Wow. 

If it wasn' already hot I'd be sweating by now... 👍 Thx a million!

To help the other few who insist on developing under Win10 / WSL / Apache/PHP and Visual Studio, these are the commands that solved it:

Make www-data the owner of the project folder XYZ:
`$ sudo chown -R www-data:www-data /mnt/c/Sites/xyz/`

Create a Symlink in /var/www/html to project folder XYZ:
`$ sudo ln -s /mnt/c/Sites/xyz/ /var/www/html/xyz`

Then I ran into an erroc in Visual Studio which was solved by
`wsl.exe --shutdown` in Powershell (Admin).

Cheers,

Carsten
